src/shared-components/dropdown/mobileDropdown.tsx

```diff
@@ -4,28 +4,28 @@ import { useTheme } from 'emotion-theming';
 import ChevronIcon from '../../svgs/icons/chevron-icon.svg';
 import { DropdownContainer, dropdownInputStyle, IconContainer } from './style';
 
-import { OptionType } from '.';
+import { OptionType, OptionValue } from '.';
 
-type MobileDropdownProps = {
+type MobileDropdownProps<T extends OptionType> = {
   borderRadius: string;
   onMobileSelectChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
-  options: OptionType[];
+  options: T[];
   textAlign: 'left' | 'center';
-  value?: string | number;
+  value?: OptionValue;
 };
 
 /**
  * The mobile dropdown is built into the `<Dropdown />` component and cannot be used individually.
  *
  * `<Dropdown /> ` will determine if the user is on a mobile device and render a true `select` tag with `option`(s).
  */
-export const MobileDropdown = ({
+export const MobileDropdown = <T extends OptionType>({
```

Comment:
I've never seen this, sort of dont understand it but I trust you. My only question is do we need this whatever it's named `<T extends OptionType>` here having declaring the same  above `type MobileDropdownProps<T extends OptionType>`
same question would apply to Desktop

Reply:
It's required in the component usage, but not the type-definition. Updated in f07da61.